packages/stratimikos/test/CMakeLists.txt

```diff
@@ -187,40 +187,70 @@ IF (${PROJECT_NAME}_ENABLE_Tpetra AND ${PACKAGE_NAME}_ENABLE_ThyraTpetraAdapters
     )
 
   TRIBITS_ADD_EXECUTABLE(
-    test_tpetra_belos_ex
-    SOURCES test_tpetra_belos_ex.cpp
+    test_tpetra_belos
+    SOURCES test_tpetra_belos.cpp
     COMM serial mpi
   )
 
   TRIBITS_ADD_TEST(
-    test_tpetra_belos_ex
-    NAME test_tpetra_belos_ex
+    test_tpetra_belos
+    NAME test_tpetra_belos
     ARGS
       "--input-file=TpetraBelosTest.xml --show-timer-summary"
     COMM serial mpi
     NUM_MPI_PROCS 4
     )
 
   TRIBITS_ADD_TEST(
-    test_tpetra_belos_ex
+    test_tpetra_belos
     NAME test_tpetra_belos_sing_reduce
     ARGS
       "--input-file=TpetraSingReduceTest.xml --show-timer-summary"
     COMM serial mpi
     NUM_MPI_PROCS 4
     )
 
+  TRIBITS_ADD_TEST(
+    test_tpetra_belos
+    NAME test_tpetra_belos_prec
+    ARGS
+      "--input-file=stratimikos_BelosPrec_ParameterList.xml --show-timer-summary"
+    COMM serial mpi
+    NUM_MPI_PROCS 4
+    )
+
+  TRIBITS_ADD_TEST(
+    test_tpetra_belos
+    NAME test_tpetra_belos_half_prec
+    ARGS
+      "--input-file=stratimikos_HalfBelosPrec_ParameterList.xml --show-timer-summary"
+    COMM serial mpi
+    NUM_MPI_PROCS 4
+    )
+
+  TRIBITS_ADD_TEST(
+    test_tpetra_belos
+    NAME test_tpetra_belos_gmres_ir
+    ARGS
+      "--input-file=stratimikos_GMRES_IR_ParameterList.xml --show-timer-summary"
+    COMM serial mpi
+    NUM_MPI_PROCS 4
+    )
+
```

Comment:
Do you need something like `Trilinos_ENABLE_FLOAT` to run these tests?

Reply:
That seems to be the case.  And you see these tests failing for builds that don't have float turned on [above](https://github.com/trilinos/Trilinos/pull/11222#issuecomment-1301375425).  (It seems that there are no PR builds that have float enabled so PR testing will not even run these tests if you guard them with `EXCLUDE_IF_NOT_TRUE ${PROJECT_NAME}_ENABLE_FLOAT` (see [`tribits_add_test()`](https://tribitspub.github.io/TriBITS/users_guide/index.html#tribits-add-test)).)

Reply:
> Do any of the nightly or master merge test builds enable float?

No, not a one.  See:

* #10742 

This means this code you are writing is not going to be tested in any automated tests that are run by the Trilinos Framework team.  The person to discuss that with is @ccober6, who is the PO for the Framework team.

Reply:
Will discuss this further with the multiprecision team. For right now, @cgcgcg says we can have this tested in the MueLu nightlies. (@csiefer2 )

Reply:
Shouldn't we at least guard these tests with `*_ENABLE_FLOAT`?

Reply:
> Shouldn't we at least guard these tests with `*_ENABLE_FLOAT`?

That is the minimum requirement.

I think you can do this by adding the argument:

```
EXCLUDE_IF_NOT_TRUE  ${PROJECT_NAME}_ENABLE_FLOAT  Tpetra_INST_COMPLEX_FLOAT
```

to the above `TRIBITS_ADD_TEST()` calls.  Using the `EXCLUDE_IF_NOT_TRUE` argument results in a line being printed to CMake output that lists the function by describes why it was not added.  (When you use an `if ()` statement, you don't see anything for a test in the output.)